cpu/stm32_common/include/periph_cpu_common.h

```diff
@@ -104,7 +104,7 @@ extern "C" {
  */
 /* Actual Lower Limit is ~100us so round up */
 #define NWDT_TIME_LOWER_LIMIT          (1U)
-#define NWDT_TIME_UPPER_LIMIT          (4U * US_PER_MS * 4096U * (1 << 6U) \
+#define NWDT_TIME_UPPER_LIMIT          (uint32_t)(4U * US_PER_MS * 4096U * (1 << 6U) \
```

Comment:
Isn't `U` for unsigned int? which if I'm not mistake us 32bit for arm-gcc, and the other variables get casted to `U` as well.

Reply:
I reverted that change and now only cast in the test application.